.eslintrc.json

```diff
@@ -0,0 +1,42 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "extends": [
+    "plugin:@typescript-eslint/recommended",
+    "prettier/@typescript-eslint",
+    "plugin:prettier/recommended"
+  ],
+  "env": {
+    "browser": true,
+    "es6": true,
+    "node": true,
+    "mocha": true,
+    "amd": true,
+    "commonjs": true
+  },
+  "plugins": [
+    "@typescript-eslint/eslint-plugin"
+  ],
+  "rules": {
+    "max-len": ["error", {"code": 120, "ignorePattern": "^import\\s.+\\sfrom\\s.+;$", "ignoreStrings": true}],
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-var-requires": "off",
+    "no-dupe-class-members": "off",
+    "@typescript-eslint/no-empty-function": ["error", { "allow": ["constructors"] }],
+    "object-curly-spacing": ["error", "always"],
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/no-use-before-define": "off"
```

Comment:
why turn off rule "no-use-before-define"?

Reply:
Before, this error occurs a lot in js file. I change the lint principle: only linting .ts files. And this "no-use-before-define" rule can be removed here.
Fixed.